packages/node_modules/@webex/plugin-meetings/test/utils/testUtils.js

```diff
@@ -191,13 +191,34 @@ const delayedTest = (callback, timeout) => new Promise((resolve) => {
   }, timeout);
 });
 
+// returns a promise that has resolve() and reject() methods
+// allowing you to resolve/reject it at any time you want
+const createControlledPromise = () => {
+  let resolvePromise, rejectPromise;
+  const promise = new Promise((resolve, reject) => {
+    resolvePromise = resolve;
+    rejectPromise = reject;
+  });
+
+  promise.resolve = resolvePromise;
+  promise.reject = rejectPromise;
+
+  return promise;
+};
+
 const addMedia = (user) => {
-  const mediaReadyEvent = {};
+  const mediaReadyPromises = {
+    local: createControlledPromise(),
+    remoteAudio: createControlledPromise(),
+    remoteVideo: createControlledPromise(),
+  };
   const mediaReady = (media) => {
     if (!media) {
```

Comment:
@marcin-bazyl 
just out of curiosity, is this a valid case when the `media:ready` callback gets an undefined media?

Reply:
this check was already there so I didn't touch it, but yeah looking at the code that sends `media:ready` it's actually not possible to ever receive no data with the event, it will always have at least `media.type` defined